builtin/providers/aws/resource_aws_autoscaling_schedule.go

```diff
@@ -0,0 +1,189 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+const AWSAutoscalingScheduleTimeLayout = "2006-01-02T15:04:05Z"
+
+func resourceAwsAutoscalingSchedule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAutoscalingScheduleCreate,
+		Read:   resourceAwsAutoscalingScheduleRead,
+		Update: resourceAwsAutoscalingScheduleCreate,
+		Delete: resourceAwsAutoscalingScheduleDelete,
+
+		Schema: map[string]*schema.Schema{
+			"arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"scheduled_action_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"autoscaling_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"start_time": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validateASGScheduleTimestamp,
+			},
+			"end_time": &schema.Schema{
```

Comment:
We could make use of the (new) resource-level [ValidateFunc](https://github.com/hashicorp/terraform/blob/master/helper/schema/schema.go#L141) to ensure that `end_time` > `start_time` here.